gossip/sync.go

```diff
@@ -283,6 +283,10 @@ func (h *handler) snapsyncStateLoop() {
 				cmd.snapsyncCancelCmd.done <- struct{}{}
 			}
 		case <-h.snapState.quit:
+			if h.snapState.cancel != nil {
```

Comment:
nit: I found that this block code appear 3 times in the `sync.go` file, so we should wrap it as a simple method like `cancelSnapsyncState()`?

```
				// cancel existing snapsync state
				if h.snapState.cancel != nil {
					_ = h.snapState.cancel()
					h.snapState.cancel = nil
				}
```